src/Monolog/Formatter/GoogleCloudLoggingFormatter.php

```diff
@@ -32,9 +36,49 @@ protected function normalizeRecord(LogRecord $record): array
         $normalized['severity'] = $normalized['level_name'];
         $normalized['time'] = $record->datetime->format(DateTimeInterface::RFC3339_EXTENDED);
 
+        // Tag with Trace ID for request attribution
+        $normalized['logging.googleapis.com/trace'] = $this->getTraceID();
+
         // Remove keys that are not used by GCP
         unset($normalized['level'], $normalized['level_name'], $normalized['datetime']);
 
         return $normalized;
     }
+
+    private function getTraceID(): ?string
+    {
+        if (empty($this->traceID) && !empty($_SERVER['HTTP_X_CLOUD_TRACE_CONTEXT'])) {
+            $matched = preg_match(
+                self::CONTEXT_HEADER_FORMAT,
+                $_SERVER['HTTP_X_CLOUD_TRACE_CONTEXT'] ?? '',
+                $matches,
+            );
+
+            if (!$matched) {
+                return null;
+            }
+
+            $projectID = $this->getProjectID();
+            if (empty($projectID)) {
+                return null;
+            }
+
+            $this->traceID = 'projects/'.$projectID.'/traces/'.strtolower($matches[1]);
+        }
+
+        return $this->traceID;
+    }
+
+    private function getProjectID(): ?string
+    {
+        if (isset($_SERVER['GOOGLE_CLOUD_PROJECT'])) {
+            return $_SERVER['GOOGLE_CLOUD_PROJECT'];
+        }
+
+        if (class_exists('\Google\Cloud\Core\Compute\Metadata')) {
```

Comment:
maybe we should add this optional dependency in `composer.json`'s `suggest`  entry (https://getcomposer.org/doc/04-schema.md#suggest)? That way user can know which package to include to get this metadata magic,

Reply:
Good idea, I'll add it now.

Reply:
Done, how is this?:

https://github.com/Seldaek/monolog/commit/05ae45ff4e0360963ab08c7d29372b2ad9f34a60#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R48